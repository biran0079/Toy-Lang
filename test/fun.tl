fun x(a,b){
  print(a);  
  print(b);  
}
x(1,2);
fun y(a){
  if(a>0){
    print(a);  
    y(a-1);
  }  
}
y(10);
fun add(a,b){
  return a+b;  
}
print(add(3,4));

