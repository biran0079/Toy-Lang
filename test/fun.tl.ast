[stmts, [fun, [id, x], [ids, [id, a], [id, b]], [stmts, [call, [id, print], [exps, [id, a]]], [tail_call, [id, print], [exps, [id, b]]]]], [call, [id, x], [exps, [int, 1], [int, 2]]], [fun, [id, y], [ids, [id, a]], [stmts, [if, [>, [id, a], [int, 0]], [stmts, [call, [id, print], [exps, [id, a]]], [tail_call, [id, y], [exps, [-, [id, a], [int, 1]]]]]]]], [call, [id, y], [exps, [int, 10]]], [fun, [id, add], [ids, [id, a], [id, b]], [stmts, [return, [+, [id, a], [id, b]]]]], [call, [id, print], [exps, [call, [id, add], [exps, [int, 3], [int, 4]]]]]]
