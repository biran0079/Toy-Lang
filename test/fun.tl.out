fun f(a)
fun g()
1
2
10
9
8
7
6
5
4
3
2
1
7
